scripts/remove_zone.py

```diff
@@ -37,73 +39,118 @@ def _is_zone_in_exchange(exchange_config_path: str) -> bool:
+            print(f"🧹 Removed {exchange.split('/')[-1]}")
         except FileNotFoundError:
             pass
 
-    # TODO: we could check if part of subZoneNames
-    # TODO: we could detect parsers that are only used by this zone
-    #       so we can clean those files up.
 
+def remove_translations(zone_key: ZoneKey):
     for locale_file in LOCALE_FILE_PATHS:
-        with JsonFilePatcher(locale_file, indent=4) as f:
+        with JsonFilePatcher(locale_file, indent=2) as f:
             zone_short_name = f.content["zoneShortName"]
             if zone_key in zone_short_name:
                 del zone_short_name[zone_key]
 
-    for api_version in ["v3", "v4"]:
+
+def remove_mockserver_data(zone_key: ZoneKey):
+    for state_level in ["daily", "hourly", "monthly", "yearly"]:
         try:
             with JsonFilePatcher(
-                ROOT_PATH / f"mockserver/public/{api_version}/state"
+                ROOT_PATH / f"mockserver/public/v6/state/{state_level}.json"
             ) as f:
                 data = f.content["data"]
-                if zone_key in data["countries"]:
-                    del data["countries"][zone_key]
+                if zone_key in data["zones"]:
+                    del data["zones"][zone_key]
 
                 for k in list(data["exchanges"].keys()):
                     if k.startswith(f"{zone_key}->") or k.endswith(f"->{zone_key}"):
                         del data["exchanges"][k]
         except FileNotFoundError:
             pass
 
-    for api_version in ["v5"]:
-        for state_level in ["daily", "hourly", "monthly", "yearly"]:
-            try:
-                with JsonFilePatcher(
-                    ROOT_PATH
-                    / f"mockserver/public/{api_version}/state/{state_level}.json"
-                ) as f:
-                    data = f.content["data"]
-                    if zone_key in data["countries"]:
-                        del data["countries"][zone_key]
-
-                    for k in list(data["exchanges"].keys()):
-                        if k.startswith(f"{zone_key}->") or k.endswith(f"->{zone_key}"):
-                            del data["exchanges"][k]
-            except FileNotFoundError:
-                pass
 
+def remove_geojson_entry(zone_key: ZoneKey):
     geo_json_path = ROOT_PATH / "web/geo/world.geojson"
-    with JsonFilePatcher(geo_json_path) as f:
+    prettier_config_path = ROOT_PATH / "web/.prettierrc.json"
+    with JsonFilePatcher(geo_json_path, indent=None) as f:
         new_features = [
             f for f in f.content["features"] if f["properties"]["zoneName"] != zone_key
         ]
         f.content["features"] = new_features
 
-    run_shell_command(f"npx prettier --write {geo_json_path}", cwd=ROOT_PATH)
+    run_shell_command(
+        f"npx prettier --config {prettier_config_path} --write {geo_json_path}",
+        cwd=ROOT_PATH,
+    )
+    run_shell_command(
+        f"pnpm generate-world",
+        cwd=ROOT_PATH / "web",
+    )
```

Comment:
This will fail if the zone we are removing is a subzone since we are not updating the parent config file.

Reply:
Hmm, which part would fail?
And yes, we are not doing anything about parent zone configs - but the "file searcher" should point out that the zone-key is used in the config and it's asking to manually clean that :)

Reply:
The generate-world script tries to look up all the zones defined as subzones in the parent config to find the "borders", since we are removing the zone from the geojson but not from the subzone field in the parent the script won't be able to find it and will crash.

I think that the crash happens before the "file searcher" so it would go unnoticed.

Reply:
Ah, now I get it! Fixed :)